wavefront_sdk/client_factory.py

```diff
@@ -0,0 +1,84 @@
+# -*- coding: utf-8 -*-
+"""Wavefront MultiClient to support Data Ingestion for Multiple Clients.
+
+@author Yogesh Prasad Kurmi (ykurmi@vmware.com)
+"""
+
+from uri import URI
+
+from wavefront_sdk.client import WavefrontClient
+from wavefront_sdk.multi_clients import WavefrontMultiClient
+
+
+# pylint: disable=W0232
+class WavefrontClientFactory:
+    """Wavefront client factory.
+
+    Create wavefront direct/proxy data ingestion client.
+    """
+
+    PROXY_SCHEME = "proxy"
+    HTTP_PROXY_SCHEME = "http"
+    DIRECT_DATA_INGESTION_SCHEME = "https"
+    clients = []
+
+    # pylint: disable=too-many-arguments
+    def add_client(self, url, max_queue_size=50000,
+                   batch_size=10000,
+                   flush_interval_seconds=5,
+                   enable_internal_metrics=True):
+        """Create a unique client."""
+        server, token = self.get_server_info_from_endpoint(url)
+
+        if self.existing_client(server):
+            raise RuntimeError("client with id " + url + " already exists.")
+        self.clients.append(WavefrontClient(server, token, max_queue_size,
+                                            batch_size, flush_interval_seconds,
+                                            enable_internal_metrics))
+
+    def get_server_info_from_endpoint(self, url):
+        """Get Server and API token from the end point.
+
+        Return None for token if URL belongs to proxy.
+        """
+        base_url = URI(url)
```

Comment:
Any benefit of using the third party over standard library `urllib.parse.urlparse`?
https://docs.python.org/3/library/urllib.parse.html#urllib.parse.urlparse

Reply:
I used this to support both python2 and python3, but as of now we are not supporting python2, I have updated the code to use the  standard library